src/imitation/util/trainer.py

```diff
@@ -40,6 +52,9 @@ def init_trainer(env_id: str,
         files, as sorted by modification times.
     policy_dir: The directory containing the pickled experts for
         generating rollouts.
+    scale: If True, then scale input Tensors to the interval [0, 1].
+    entropy_weight: Only applicable for AIRL. The entropy_weight argument of
```

Comment:
Can't this be part of `discrim_kwargs`?

Reply:
The GAIL discrim net doesn't have entropy_weight right now, so this breaks GAIL

Reply:
Ah, we could just add an analogous entropy weight here:
![image](https://user-images.githubusercontent.com/1750835/62153824-4999fb00-b2ba-11e9-84d7-6e3f59121f7c.png) . I'll do this as part of the PR


There's a second entropy weight (\lambda in eq 18) that I'll want to add in later because some tuning this was important for Reacher results.
![image](https://user-images.githubusercontent.com/1750835/62153811-41da5680-b2ba-11e9-90a5-a5cceffd1486.png)

Reply:
Actually, I'm not even sure if this is analogous/useful, because the DiscrimNetGAIL entropy weight would go in `DiscrimNetGAIL.discriminator_loss`, whereas the DiscrimNetAIRL entropy weight is in `DiscrimNetAIRL.build_policy_train`.

Let me know what you think

Reply:
If there was a direct analogue of AIRL's `entropy_weight` in GAIL then it would make more sense to put it at the top-level, although I'm sitll hesitant about making too many things top-level configuration parameters (what if we want `train` to be used for something other than AIRL/GAIL in the future?) I'm not that familiar with GAIL but I think you're right there are similar but different concepts, which should probably have different names.

Configuration parameters that are silently ignored under some settings can get confusing quickly. Originally I was imagining that a default `entropy_weight` would get set by e.g. the `airl` named config. I now understand that you might want to set it on a per-environment basis. The options I see are:
  - 1. Use current approach of a top level config parameter that is ignored under GAIL, but add documentation explaining this.
  - 2. Set it in `discrim_kwargs`, but in the environment named config check that `use_gail` is False before setting it. This is self-documenting, but would break if you executed the environment named config before the AIRL config.
  - 3. Make `discrim_kwargs` and `reward_kwargs` be a dict of algorithm names to parameters, with a `'default'` key that applies to all algorithms. The environment named config can then just set `discrim_kwargs['airl']['entropy_weight'] = 1.0`.
I think right now I favour solution (3), but none of them seem great.

Reply:
thanks for the ideas. (iii) seems like the best long term solution but also crufty

Reply:
I think I will do something like (iii) if we get another algorithm-specific variable like `entropy_weight` -- (i) seems not-too-bad to me right now.  Adding in a sentence or two of documentation and renaming variable to `airl_entropy_weight` for now.